ui/tests/integration/components/project-input-variables-list-test.ts

```diff
@@ -0,0 +1,123 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import { create, collection, clickable, isPresent, fillable, text } from 'ember-cli-page-object';
+import { Ref, GetProjectRequest } from 'waypoint-pb';
+
+
+const page = create({
+  hasForm: isPresent('[data-test-input-variables-form]'),
+  variablesList: collection('[data-test-input-variables-list-item]', {
+    dropdown: clickable('[data-test-input-variables-dropdown]'),
+    dropdownEdit: clickable('[data-test-input-variables-dropdown-edit]'),
+    dropdownDelete: clickable('[data-test-input-variables-dropdown-delete]'),
+    isHcl: isPresent('[data-test-input-variables-list-item-is-hcl]'),
+    varName: text('[data-test-input-variables-var-name]'),
+    varValue: text('[data-test-input-variables-var-value]'),
+  }),
+  createButton: clickable('[data-test-input-variables-add-variable]'),
+  cancelButton: clickable('[data-test-input-variables-edit-cancel]'),
+  saveButton: clickable('[data-test-input-variables-edit-save]'),
+  varName: fillable('[data-test-input-variables-var-name]'),
+  varStr: fillable('[data-test-input-variables-var-str]'),
+});
+
+module('Integration | Component | project-input-variables-list', function (hooks) {
+  setupRenderingTest(hooks);
+  setupMirage(hooks);
+
+  test('it renders', async function (assert) {
+    let dbproj = await this.server.create('project', 'with-input-variables', { name: 'Proj1' });
+
+    let api = this.owner.lookup('service:api');
+    let ref = new Ref.Project();
+    ref.setProject(dbproj.name);
+    let req = new GetProjectRequest();
+    req.setProject(ref);
+
+    let resp = await api.client.getProject(req, api.WithMeta());
+    let project = resp.getProject();
+    this.set('project', project.toObject());
```

Comment:
While this probably is the most-accurate way to create the same shape of object that the real app will deal with, I think for the purposes of this test we can skip the API layer:

```suggestion
    let dbproj = await this.server.create('project', 'with-input-variables', { name: 'Proj1' });
    let project = dbproj.toProtobuf();
    this.set('project', project.toObject());

```

Reply:
I'm personally on the fence on this one (I like the accuracy of using the API), but I'll go with your suggestion.